shared/util/use-focus-blur.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react'
+import {useNavigationEvents} from './navigation-hooks'
+import {NavigationEventCallback} from '@react-navigation/core'
+
+/**
+ * useFocusBlur sets up callbacks for when your component is focused or blurred.
+ * These happen when you are mounted/unmounted, and when navigation focus and
+ * blur events happen. onFocus and onBlur should be memoized, if not they may be
+ * called more than you expect.
```

Comment:
Maybe add a note here about how this won't work outside the navigation context?

Reply:
It should work, I put `useNavigationEvents` in a try/catch. Will test

Reply:
Oh, I didn't look very carefully at the `useEffect`, oops